### DISCUSSION TITLE
chapter_searching/

### DISCUSSION THREAD

**Discussion by giscus** [krahets/hello-algo#540] [Announcements]
# chapter_searching/

一本动画图解、能运行、可提问的数据结构与算法入门书

https://www.hello-algo.com/chapter_searching/

**Comment by 5710xIa** [comment]
突然很好奇，人的脑子是怎么完成搜索自己的记忆的呢？

**Reply by volmodaoist** [reply]
我觉得人脑里面应该是有类似于 LRU，LFU 缓存机制的区域的

**Reply by krahets** [reply]
@volmodaoist 很有趣，感觉缓存机制和大脑的渐层记忆区和深层记忆区的工作机制有些相似点

**Reply by github15238484509** [reply]
你们说的啥呀，我一个大专生听不懂呀，脑袋就快炸了7 8 9三章我看了2天才看完

**Reply by Gerhart1998** [reply]
LRU：最近最少使用（Least Recently Used, LRU）：根据数据项最近被访问的时间，淘汰最久未被使用的数据项。
 LFU：最少使用（Least Frequently Used, LFU）：根据数据项被访问的频率，淘汰访问次数最少的数据项。

**Comment by sanbei101** [comment]
好难啊

**Reply by krahets** [reply]
加油啊！

**Comment by Yerosius** [comment]
为什么这章不叫做查找

**Reply by krahets** [reply]
Hi，看一下“10.5   重识搜索算法”，里面有做分类

**Comment by xuanxuanxie** [comment]
好难555555

**Comment by Autumn00air** [comment]
这幅图，初中的某一个周末的下午，做过一个梦，就是在这样的一个图书馆里不断地走，发自内心的开心和满足的感觉。印象深是因为醒来的时候一家人都在笑我睡着的时候在嘿嘿傻笑，问我梦见什么了这么梦里都能笑出来，我把这个梦和他们说了，我爸笑着说，好！我们家怎么也得缴起一个大学生出来！我希望我能一直热爱读书。

**Comment by ffmm520** [comment]
封面图让我想起了纪念碑谷

**Comment by lexign** [comment]
不得不说，作者每个章节的配图都很Nice！

**Comment by xiaozhuge123** [comment]
day08

**Comment by zhengads** [comment]
2025.11.27 先跳过树堆图，先学些算法

**Reply by JensenLei123** [reply]
太痛了 树堆图学的似懂似不懂的状态 概念感觉好 但是在算法实现上就有点。。。。希望有内化点什么知识吧。。。毕竟反复耗死在一个章节反复啃，倒不如先试着学一遍然后往前学，等到在其他地方又遇到树堆图的时候，通过不断的接触接触再接触直到理解

**Comment by Dumbliidore** [comment]
跳过了堆过来看的
